### PR TITLE
Api actions

### PR DESCRIPTION
**Pull request by Casey1449**

### CODE REVIEW

**Comment by Casey1449** [comment]
This is some preliminary pseudo-coding. Probably not too dangerous to merge, but will have to be fine-tuned

**Comment by swanie21** [comment]
We want to do something with the 'weater' data!

This is a good starting point to work with grabbing the api data with redux, nice job setting this up Casey!

File: src/actions/index.js [hunk 1]
```diff
@@ -1,9 +1,14 @@
 export const RECEIVE_FORECAST = 'RECEIVE_FORECAST'
 
 export const receiveForecast = (forecastData) => ({
-  // action object
+  type: RECEIVE_FORECAST,
+  forecastData,
+  data: do something to the weater data
```
Comment (stevekinney) [inline src/actions/index.js:6]: Does this compile?

File: src/actions/index.js [hunk 2]
```diff
@@ -1,9 +1,14 @@
...
 });
 
-export const fetchForecast = options => dispatch => {
-  // return fetch API call
+export const fetchZipForecast = forecastType, zip, key => dispatch => {
```
Comment (stevekinney) [inline src/actions/index.js:9]: I'm in love this this function.

Comment (stevekinney) [reply src/actions/index.js:9]: Hey @joshuajhun, check out this function!

Comment (Casey1449) [reply src/actions/index.js:9]: 🤗 
But I think we should pull the 'key' variable back outside the function, where it makes sense to live as a thing that gets passed in, no? (The thinking was if the key ever expires, it'll only need to be updated in one place.)

**Comment by stevekinney** [comment]
You still want to pass in the key, that way it's not reliant on global state
and if you ever needed to unit test it, you could.

**Pull request merged**